src/frontend/index.js

```diff
@@ -1,21 +1,31 @@
 requirejs.config({
-    baseUrl: '.',
+    packages: [
+        { name: 'when', location: '../bower_components/when', main: 'when' },
+        { name: 'mod_engine', location: '../backend/engine', main: 'index'},
+        { name: 'mod_executor', location: '../backend/executor', main: 'index'},
+        { name: 'mod_process', location: '../backend/process', main: 'index'},
+        { name: 'mod_data_structures', location: '../backend/data_structures', main: 'index'},
+        { name: 'mod_parser', location: '../backend/parser', main: 'index'}
+    ],
     paths: {
-        lib: '../bower_components',
-        module: '../backend'
+        backend: "../backend/index"
     }
 });
 
-var backend;
-
-function initBackend() {
-    require(["backend/index"], function(be) {
-        backend = be;
+function runSampleProgram() {
+    require(["backend"], function(backend) {
+        backend.runProgram("x=2")
```

Comment:
backend will store statically the program code / compiled code and execution state? That mean I can call the backend in separate function and it will still use the previous code?